app/services/target_service/compatible_target_finder.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module TargetService
+  class CompatibleTargetFinder
+    def initialize(target)
+      @target = target
+      @user = target.user
+      @topic = target.topic
+    end
+
+    def call
+      find_compatible_targets
+    end
+
+    private
+
+    def match_targets
+      @match_targets ||= Target.where.not(user: @user).where(topic: @topic)
+    end
+
+    def find_compatible_targets
+      match_targets.each do |match_target|
+        distance = Geocoder::Calculations.distance_between([@target.latitude, @target.longitude],
+                                                           [match_target.latitude, match_target.longitude])
+        if distance <= @target.radius + match_target.radius
```

Comment:
Why is this sum `@target.radius + match_target.radius`?

Reply:
* In order to validate the intersection, the difference between these two points should be less or equal to the sum of their radius.

Reply:
@JulianPasquale I will implement the [geokit-rails gem](https://github.com/geokit/geokit-rails) suggested in this comment by @sebastiancaraballo. So this code will be updated!